plugins/catalog-react/src/hooks/useEntityPermission.ts

```diff
@@ -46,10 +65,11 @@ export function useEntityPermission(permission: Permission): {
   );
 
   if (loadingEntity || loadingPermission) {
-    return { loading: true, allowed: false };
+    return { loading: true, allowed: defaultValue };
   }
   if (entityError) {
     return { loading: false, allowed: false, error: entityError };
   }
+  cache.current[`${permission.name}${stringifyEntityRef(entity)}`] = allowed;
```

Comment:
Hi! Please help me understand this 😅  Here `useEntity` is fetching some entity coming from some context (instantiated somewhere). The entity, it isn't going to change unless the user navigates to another route, pointing to another entity. In that case, the component using `useEntityPermission` is going to be unmounted and the hook will be instantiated again with a new state (means the cache will be gone).
So the question is: what are we caching here? 😂
Also, if we have two components invoking useEntityPermission (with the same permission and same entity) one after the other, nothing will be cached.

Reply:
Good catch, I confused myself over the usage of `useRef` - for some reason I had convinced myself it keeps a mutable value across renders AND components but in reality it's just across renders. I've changed the cache to be a plain object at the file level.